utils/mount/mount_service.go

```diff
@@ -123,7 +123,7 @@ func NewMountService(target, upper string, lowLayers []string) (Service, error)
 //NewMountServiceByTarget will filter file system by target,if not existed,return false.
 func NewMountServiceByTarget(target string) Service {
 	mounted, info := GetMountDetails(target)
-	if !mounted {
+	if !mounted || !strings.Contains(info.Work, target) {
```

Comment:
I'm confused that why workdir will contains the target? I guess they are not relevant?

Reply:
Now sealer will list all dirs in the /var/lib/sealer/tmp like this
![image](https://user-images.githubusercontent.com/16217209/189845353-a9229aee-76da-43bf-b97c-59f93448da27.png)

And sealer mount the path like this:
```golang
func (o *Overlay2) Mount(target string, upperLayer string, layers ...string) error {
	if target == "" {
		return fmt.Errorf("target cannot be empty")
	}
	if len(layers) == 0 {
		return fmt.Errorf("layers cannot be empty")
	}
	workdir := path.Join(target, "work")
	if err := os.MkdirAll(workdir, os.ModePerm); err != nil {
		return fmt.Errorf("failed to create workdir: %v", err)
	}
	var err error
	defer func() {
		if err != nil {
			_ = os.RemoveAll(workdir)
		}
	}()

	var indexOff string
	// figure out whether "index=off" option is recognized by the kernel
	_, err = os.Stat("/sys/module/overlay/parameters/index")
	switch {
	case err == nil:
		indexOff = "index=off,"
	case os.IsNotExist(err):
		// old kernel, no index -- do nothing
	default:
		logrus.Warnf("Unable to detect whether overlay kernel module supports index parameter: %s", err)
	}

	mountData := fmt.Sprintf("%slowerdir=%s,upperdir=%s,workdir=%s", indexOff, strings.Join(strUtils.Reverse(layers), ":"), upperLayer, workdir)
	logrus.Debugf("mount data : %s", mountData)
	if err = mount("overlay", target, "overlay", 0, mountData); err != nil {
		return fmt.Errorf("failed to create overlay mount to %s: %v", target, err)
	}
	return nil
}
```